packages/@aws-cdk/core/lib/cfn-parse.ts

```diff
@@ -117,6 +118,18 @@ export class FromCloudFormation {
         value: tag.Value,
       };
   }
+
+  public static getTypeUnion(validators: Validator[], mappers: Mapper[], value: any): any {
```

Comment:
Feels like this function would be better off taking an array of pairs, rather than a pair of arrays?

Reply:
I modeled this after the existing [`unionMapper` function in `runtime.ts`](https://github.com/aws/aws-cdk/blob/3a53242231a0545f41bf6f77d8ac044df0836152/packages/%40aws-cdk/core/lib/runtime.ts#L86-L86).

Making it an array of pairs would make it much more difficult to write code like this:

```ts
        const validatorNames = types.map(type => genspec.validatorName(type).fqn).join(', ');

        const arg = `prop${this.depth}`;
        const mappers = types
          .map(type => `(${arg}: any) => ${this.deeperCopy(arg).visitAtom(type)}`)
          .join(', ');

        return `${CFN_PARSE}.FromCloudFormation.getTypeUnion([${validatorNames}], [${mappers}], ${this.baseExpression})`;

```

With the current modeling, you can easily generate separate validators and mappers. Making `getTypeUnion ()` take an array of pairs would make this a fair bit more complicated.